src/renderer/src/routes/modals/ExportFullTransactions/index.tsx

```diff
@@ -75,8 +78,14 @@ const ExportFullTransactionsModal = () => {
         dateTo: (dateFns.isSameDay(today, actionData.to) ? today : actionData.to).toJSON(),
       })
 
-      const formattedDateFrom = dateFns.format(actionData.from, t('filenameDateFormat'))
-      const formattedDateTo = dateFns.format(actionData.to, t('filenameDateFormat'))
+      const formattedDateFrom = DateHelper.formatLocalized(actionData.from, {
+        language,
+        format: t('filenameDateFormat'),
+      })
+      const formattedDateTo = DateHelper.formatLocalized(actionData.to, {
+        language,
+        format: t('filenameDateFormat'),
+      })
```

Comment:
What do you think? And remove `filenameDateFormat`.
```suggestion
      const format = 'MMddyyyy'
      const formattedDateFrom = DateHelper.formatLocalized(actionData.from, {
        language,
        format,
      })
      const formattedDateTo = DateHelper.formatLocalized(actionData.to, {
        language,
        format,
      })
```